### PR TITLE
[Hub Generated] Review request for Microsoft.KeyVault to add version preview/2021-06-01-preview

### PR DESCRIPTION
**Pull request by susolank**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/susolank/azure-rest-api-specs/dev-keyvault-Microsoft.KeyVault-2021-06-01-preview...main/).

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-06-01-preview/keys.json [hunk 1]
```diff
@@ -423,6 +423,10 @@
         "rotationPolicy": {
           "$ref": "#/definitions/RotationPolicy",
           "description": "Key rotation policy in response. It will be used for both output and input. Omitted if empty"
+        },
+        "releasePolicy": {
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-06-01-preview/keys.json:427]: If it's a goal to match the data-plane, this should be "release_policy": https://github.com/Azure/azure-rest-api-specs/blob/ee1dc806b00b73458a7d0de7b05da3c56c756cfb/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json#L1556-L1560.

Similarly, I recommend using `KeyReleasePolicy` for the model definition to be consistent. Might make troubleshooting easier, as well as dev UX.

**Review by heaths — APPROVED** [review]
One change to consider, but I won't block the PR based solely on that.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-06-01-preview/keys.json (hunk 1 above)
Comment (susolank) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-06-01-preview/keys.json]: @heaths Made suggested changes. Please review PR again.

Comment (susolank) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-06-01-preview/keys.json]: Lint Checker is complaining that "release_policy" should be camelCase. Would this change be acceptable? 

<html>
<body>


❌ R3016 - DefinitionsPropertiesNamesCamelCase | Property named: 'release_policy', for definition: 'KeyProperties' must follow camelCase style. Example: 'release_policy'.Location: Microsoft.KeyVault/preview/2021-06-01-preview/keys.json#L427
-- | --



</body>
</html>

**Review by dalzhao — APPROVED** [review]

**Review by heaths — APPROVED** [review]

**Comment by susolank** [comment]
@ruowan  I feel this breaking change review process is not applicable to this PR because new properties are being added the KeyCreateParameters object. No existing properties are being modified or removed. No existing customers will be impacted by this change. This change is allow customers to create Secure Key Release (SKR) keys using ARM template.

**Comment by susolank** [comment]
<html>
<body>


Rule | Message
-- | --
❌ 1041 - AddedPropertyInResponse | The new version has a new property 'release_policy' in response that was not found in the old version.New: Microsoft.KeyVault/preview/2021-06-01-preview/keys.json#L354:7Old: Microsoft.KeyVault/preview/2021-06-01-preview/keys.json#L354:7
❌ 1045 - AddedOptionalProperty | The new version has a new optional property 'release_policy' that was not found in the old version.New: Microsoft.KeyVault/preview/2021-06-01-preview/keys.json#L354:7Old: Microsoft.KeyVault/preview/2021-06-01-preview/keys.json#L354:7
❌ 1045 - AddedOptionalProperty | The new version has a new optional property 'exportable' that was not found in the old version.New: Microsoft.KeyVault/preview/2021-06-01-preview/keys.json#L436:7Old: Microsoft.KeyVault/preview/2021-06-01-preview/keys.json#L432:7
💬 1001 - NoVersionChange | The versions have not changed.New: Microsoft.KeyVault/preview/2021-06-01-preview/keys.json#L1:1Old: Microsoft.KeyVault/preview/2021-06-01-preview/keys.json#L1:1


</body>
</html>

**Comment by heaths** [comment]
@lmazuel can you help merge this? As with other KV REST PRs, the remaining issues are, unfortunately, expected.

**Comment by ruowan** [comment]
There are no cusomer impact. Here is the reason for adding new property breaking change: https://github.com/Azure/azure-rest-api-specs/pull/16042#issuecomment-919542336.

**Comment by ruowan** [comment]
@JeffreyRichter could you please help to review this breaking change?

**Review by ruowan — APPROVED** [review]

**Pull request merged**